controllers/market/updater.js

```diff
@@ -0,0 +1,157 @@
+const responses = require('../../utils/responses/responses');
+
+const bidUpdate = async (req, res) => {
+    const userId = req.user.id.id;
+    const eventId = req.eventId;
+    const bidId  = req.params.bidId;
+    let { value, marketId, isDirectPurchase = false } = req.body;
+    value = parseInt(value);
+    if (value == null || marketId == null) {
+        return responses.errorDTOResponse(res, 400, 'Debe proporcionar un valor y un marketId');
+    }
+
+    const market = await Market.findOne({
+        raw: true,
+        where: {
+            [Op.and]: [{
+                eventId: eventId
+            }, {
+                id: marketId
+            }]
+        }
+    });
+    const bid = await Bid.findOne({
+        raw: true,
+        where: {
+            [Op.and]: [{
+                userId: userId
+            },{
+                marketId: marketId
+            }, {
+                id: bidId
+            }]
+        }
+    });
+    const player = await PlayerFantasy.findOne({
+        raw: true,
+        where: {
+            [Op.and]: [
+            {
+                userId: userId
+            }, {
+                eventId: eventId
+            }]
+        }
+    })
+
+    if (!bid) {
+        return responses.errorDTOResponse(res, 400, 'No existe esta oferta');
+    }
+
+    if (!player) {
+        return responses.errorDTOResponse(res, 400, 'No estas participando en este evento');
+    }
+    if (!market) {
+        return responses.errorDTOResponse(res, 400, 'No existe esta subasta');
+    }
+    if (player.money < value) {
+        return responses.errorDTOResponse(res, 400, 'No tienes suficiente dinero');
+    }
+    if (market.isFinished) {
+        return responses.errorDTOResponse(res, 400, 'La subasta ya ha finalizado');
+    }
+    if (bid.value > value) {
+        return responses.errorDTOResponse(res, 400, 'El valor de la oferta debe ser mayor al valor inicial');
+    }
+    if (market.userId == userId) {
+        return responses.errorDTOResponse(res, 400, 'No puedes ofertar en tu propia subasta');
+    }
+    if (market.immediatePurchaseValue == value && isDirectPurchase) {
+        const warehouse = await Warehouse.findOne({
+            raw: true,
+            where: {
+                [Op.and]: [{
+                    stickerId: market.stickerId
+                }, {
+                    eventId: eventId
+                }, {
+                    userId: userId
+                }]
+            }
+        });
+        if (!warehouse) {
+            await Warehouse.create({
+                quantity: 1,
+                stickerId: market.stickerId,
+                userId: userId,
+                eventId: eventId
+            });
+        } else {
+            await warehouse.update({
+                quantity: warehouse.quantity + 1
+            });
+        }
+        await PlayerFantasy.update({
+            money: player.money - value
```

Comment:
Creo que esto es inncesario, cuando es compra directa podemos omitir el value pt2
```suggestion
            money: player.money - market.immediatePurchaseValue
```